sentry-samples/sentry-samples-console/src/main/java/io/sentry/samples/console/Main.java

```diff
@@ -0,0 +1,132 @@
+
+          // Allows inspecting and modifying, returning a new or simply rejecting (returning null)
+          options.setBeforeBreadcrumb(
+              (breadcrumb, hint) -> {
+                // Don't add breadcrumbs with message containing:
+                if (breadcrumb.getMessage() != null
+                    && breadcrumb.getMessage().contains("bad breadcrumb")) {
+                  return null;
+                }
+                return breadcrumb;
+              });
+
+          // Configure the background worker which sends events to sentry:
+          // Wait up to 5 seconds before shutdown while there are events to send.
+          options.setShutdownTimeout(5000);
+
+          // Enable SDK logging with Debug level
+          options.setDebug(true);
+          // To change the verbosity, use:
+          options.setDiagnosticLevel(SentryLevel.LOG);
+
+          // Using a proxy:
+          options.setProxy(null); // new Proxy(Type.HTTP, new InetSocketAddress(8090)));
+
+          // Exclude classes from certain packages from stack traces sent to Sentry:
+          options.addInAppExclude("org.jboss");
+        });
+
+    Sentry.addBreadcrumb(
+        "A 'bad breadcrumb' that will be rejected because of 'BeforeBreadcrumb callback above.'");
+
+    // Data added to the root scope (no PushScope called up to this point)
+    // The modifications done here will affect all events sent and will propagate to child scopes.
+    Sentry.configureScope(
+        scope -> {
+          scope.addEventProcessor(new SomeEventProcessor());
+
+          scope.setExtra("SomeExtraInfo", "Some value for extra info");
+        });
+
+    // Configures a scope which is only valid within the callback
+    Sentry.withScope(
+        scope -> {
+          scope.setLevel(SentryLevel.FATAL);
+          scope.setTransaction("main");
+
+          Sentry.captureMessage("Fatal message!");
+        });
+
+    Sentry.captureMessage("Some warning!", SentryLevel.WARNING);
+
+    Exception exception = new RuntimeException("Attempting to send this multiple times");
+
+    // Only the first capture will be sent to Sentry
+    for (int i = 0; i < 3; i++) {
+      // The SDK is able to detect duplicate events:
+      // This is useful, for example, when multiple loggers log the same exception. Or exception is
+      // re-thrown and recaptured.
+      Sentry.captureException(exception);
+    }
+
+    int count = 10;
+    for (int i = 0; i < count; i++) {
+      String msg = "%d of %d items we'll wait to flush to Sentry!";
+      Message message = new Message();
+      message.setMessage(msg);
+      message.setFormatted(String.format(msg, i, count));
+      SentryEvent event = new SentryEvent();
+      event.setMessage(message);
+      Sentry.captureEvent(event, SentryLevel.DEBUG);
+    }
+
+    // Console output will show queue being flushed. Task completes then and timeout is never
+    // reached (you don't need to wait a day :)
+    Sentry.flush(10000);
```

Comment:
Not sure this will work. It should but we need to work on `flush`

Reply:
You're right - it doesn't. `SentryClient#flush` is literally `// TODO: Flush transport` :-)

Reply:
yep, I'd rather keep `flush` out of this sample, but change it to `close` which is already implemented?

Reply:
Since `close` is executed anyway at the JVM termination I am not sure if this wouldn't be confusing. I'll leave it up to you and @bruno-garcia. This is anyway not the final version of the sample as it will get likely updated once more features are added to Java SDK.

Reply:
that's true, if the integration call closes automatically by default, no need, or, to still sample it, but add as a comment that a default integration will do it, but they still could do it manually. @bruno-garcia ?

Reply:
Addressed in #502.